src/other_functions.jl

```diff
@@ -178,3 +179,70 @@ polynomials. For more details, see [`Base.isapprox`](@ref).
 function isapprox{T<:AbstractSeries,S<:AbstractSeries}(x::T, y::S; rtol::Real=rtoldefault(x,y), atol::Real=0, nans::Bool=false)
     x == y || (isfinite(x) && isfinite(y) && norm(x-y,1) <= atol + rtol*max(norm(x,1), norm(y,1))) || (nans && isnan(x) && isnan(y))
 end
+
+
+#taylor_expand function for Taylor1
+doc"""
+    taylor_expand(f,x0;order)
+
+Makes a Taylor expansion of the function `f` around the point `x0`. If x0 is a scalar,
+a `Taylor1` expansion will be done. If `x0` is a vector, a `TaylorN` expansion will be
+computed, changing the number of `TaylorN` variables
+according to the dimension of `f`.
+"""
+function taylor_expand(f::Function; order::Int64=15)
+   a = Taylor1(order)
+   return f(a)
+end
+
+function taylor_expand{T<:Number}(f::Function, x0::T; order::Int64=15)
+   a = Taylor1([x0, one(T)], order)
+   return f(a)
+end
+
+#taylor_expand function for TaylorN
+function taylor_expand{T<:Number}(f::Function, x0::Vector{T}; order::Int64=get_order()) #a Taylor expansion around x0
```

Comment:
I agree that, for `Taylor1` expansions it makes sense to allow the user to set the `order` of the expansion; I am not in favor of this for `TaylorN` expansions, at least as it is now. My objection is related to the use of `set_variables`, since it changes some internals, in particular, the *maximum* order of the `TaylorN` expansions. This costs time; also, the idea of the current implementation is to set the parameters related to TaylorN internals only once at the very beginning.

I think this method should assert that the the length of `x0` corresponds to `get_numvars()`, as you do it, and *only* when this is fulfilled evaluate `f(X .+ x0)` (notice the spaces). In this case, using `set_variables` will not change anything internally.

An alternative that avoids using `set_variables` is for example:
```julia
function taylor_expand{T<:Number}(f::Function, x0::Vector{T}; order::Int64=get_order())
    ll = length(x0)
    # check that x0 has the correct length and `order` does not exceed the maximum order of the expansions
    @assert ll == get_numvars() && order <= get_order()

    X = Array{TaylorN{T}}(length(x0)
    for i in eachindex(X)
        X[i] = TaylorN(T, i, order=order)
    end

    return f( X .+ x0 )
end
```

This alternative permits to have expansions of given `order`, smaller or equal to `get_order()`, e.g., `order=2` when `get_order()` is 6.

Reply:
Really in favor of this. By the way, really good "type preserving strategy" doing the `X[i] = TaylorN(T, i, order)`.

Reply:
Regarding the problem with travis in Julia 0.5 (and considering [your comment](https://github.com/JuliaDiff/TaylorSeries.jl/pull/121#issuecomment-334968066) about this), I think a possible solution is to have `X` as the shifted vector. I mean, define `X[i] = x0[i] + TaylorN(T, i, order=order)` *inside* the for-loop, and then only return the evaluation of `f( X )`. This could be used in each method of `taylor_expand`.

I think this could actually solve the problem with travis, and yield a tiny performance improvement, since only one loop would be needed.